Controllers/controller.py

```diff
@@ -106,3 +111,20 @@ def open_file_dialog(self):
             url = file_dialog.selectedUrls()[0]
             self._media_player.setSource(url)
             self._media_player.play()
+
+    @Slot()
+    def position_changed(self):
+        """Command the video player to change state of the position"""
+        position = self._media_player.position()
+        self._window.media_panel.progress_bar_slider.setValue(position)
+
+    @Slot()
+    def duration_changed(self):
+        """Command the video player to change the state when duration changed"""
```

Comment:
This comment seems inaccurate. This function is changing the state of the progress bar, not the video player. Here is perhaps a better comment:
"""
Sets the range of the progress bar when the duration of the media player changes.
"""